lib/util/s3.ts

```diff
@@ -5,7 +5,12 @@ import { parseUrl } from './url';
 let s3Instance: S3 | undefined;
 export function getS3Client(): S3 {
   if (!s3Instance) {
-    s3Instance = new S3({});
+    const endpoint = process.env.RENOVATE_X_S3_ENDPOINT;
+    const forcePathStyle = process.env.RENOVATE_X_S3_PATH_STYLE;
+    s3Instance = new S3({
+      ...(endpoint && { endpoint }),
+      ...(forcePathStyle && { forcePathStyle: true }),
+    });
```

Comment:
Please add a test to cover all branches:
![image](https://user-images.githubusercontent.com/1798109/186614357-b3f720ad-7399-4b6b-bda1-1142ede05db7.png)

Reply:
on second thought ill do it here, as its requires me to extract changes from this PR and updating docs

Reply:
should be done here
https://github.com/renovatebot/renovate/blob/88c9ca6aa37aa56e785ab3b451d394f5c3126999/lib/util/s3.spec.ts#L1

there is no mock, but you should use dynamic import in test, so you always get a fresh module from jest. we do this here for example

https://github.com/renovatebot/renovate/blob/88c9ca6aa37aa56e785ab3b451d394f5c3126999/lib/modules/platform/bitbucket-server/index.spec.ts#L209

Reply:
```diff
diff --git a/lib/util/s3.spec.ts b/lib/util/s3.spec.ts
index 09afe4967..31929e296 100644
--- a/lib/util/s3.spec.ts
+++ b/lib/util/s3.spec.ts
@@ -1,6 +1,12 @@
 import { getS3Client, parseS3Url } from './s3';

 describe('util/s3', () => {
+  afterEach(() => {
+    delete process.env.RENOVATE_X_S3_ENDPOINT;
+    delete process.env.RENOVATE_X_S3_PATH_STYLE;
+    jest.resetModules();
+  });
+
   it('parses S3 URLs', () => {
     expect(parseS3Url('s3://bucket/key/path')).toEqual({
       Bucket: 'bucket',
@@ -9,7 +15,7 @@ describe('util/s3', () => {
   });

   it('returns null for non-S3 URLs', () => {
-    expect(parseS3Url('http://example.com/key/path')).toBeNull();
+    expect(parseS3Url(new URL('http://example.com/key/path'))).toBeNull();
   });

   it('returns null for invalid URLs', () => {
@@ -21,4 +27,21 @@ describe('util/s3', () => {
     const client2 = getS3Client();
     expect(client1).toBe(client2);
   });
+
+  it('is uses experimental env', async () => {
+    process.env.RENOVATE_X_S3_ENDPOINT = 'https://minio.domain.test';
+    process.env.RENOVATE_X_S3_PATH_STYLE = 'true';
+    const s3 = await import('./s3');
+    const client1 = s3.getS3Client();
+    const client2 = getS3Client();
+    expect(client1).not.toBe(client2);
+    expect(await client1.config.endpoint()).toStrictEqual({
+      hostname: 'minio.domain.test',
+      path: '/',
+      port: undefined,
+      protocol: 'https:',
+      query: undefined,
+    });
+    expect(client1.config.forcePathStyle).toBeTrue();
+  });
 });
```
Please add to this PR 😉